lib/ansible/plugins/action/__init__.py

```diff
@@ -325,31 +325,35 @@ def _fixup_perms(self, remote_path, remote_user, execute=False, recursive=True):
                         # fs acls failed -- do things this insecure way only
                         # if the user opted in in the config file
                         self._display.warning('Using world-readable permissions for temporary files Ansible needs to create when becoming an unprivileged user which may be insecure. For information on securing this, see https://docs.ansible.com/ansible/become.html#becoming-an-unprivileged-user')
-                        self._remote_chmod('a+%s' % mode, remote_path, recursive=recursive)
+                        self._remote_chmod('a+%s' % mode, remote_path, recursive=recursive, raise_on_error=True)
                     else:
                         raise AnsibleError('Failed to set permissions on the temporary files Ansible needs to create when becoming an unprivileged user. For information on working around this, see https://docs.ansible.com/ansible/become.html#becoming-an-unprivileged-user')
         elif execute:
             # Can't depend on the file being transferred with execute
             # permissions.  Only need user perms because no become was
             # used here
-            self._remote_chmod('u+x', remote_path, recursive=recursive)
+            self._remote_chmod('u+x', remote_path, recursive=recursive, raise_on_error=True)
 
         return remote_path
 
-    def _remote_chmod(self, mode, path, recursive=True, sudoable=False):
+    def _remote_chmod(self, mode, path, recursive=True, sudoable=False, raise_on_error=False):
```

Comment:
Sorry, I think we had a misunderstanding when we talked earlier.  I don't think we should be raising unless we add more exceptions so that we can tell why an exception occurred (right now, we'd have to catch the AnsibleError and then parse the message to tell why we failed.)

raise_on_error is bad API.  The API should either raise whenever there is an error or let the caller discriminate.  Passing in a flag to tell the function to raise isn't meaningful.

If we start raising an error, then we have to audit the code and decide what the failure case means in the present code.  If the code doesn't depend on it (or works in some scenarios) then we probably have to replicate that behaviour instead of changing to always failing.